src/main/java/net/minecraftforge/common/ISpecialArmor.java

```diff
@@ -139,11 +156,11 @@ else if (stack.getItem() instanceof ItemArmor && !source.isUnblockable())
                     }
                     ratio += prop.AbsorbRatio;
 
-                    double absorb = damage * prop.AbsorbRatio;
+                    double absorb = damage * prop.AbsorbRatio + (totalArmor > 0 ? absorbedDamage * prop.Armor / totalArmor : 0);
                     if (absorb > 0)
                     {
-                        ItemStack stack = inventory[prop.Slot];
-                        int itemDamage = (int)(absorb / 25D < 1 ? 1 : absorb / 25D);
+                        ItemStack stack = inventory[prop.Slot.getIndex()];
+                        int itemDamage = (int)(absorb < 1 ? 1 : absorb);
                         if (stack.getItem() instanceof ISpecialArmor)
```

Comment:
Issue is how does one claim to take all of the damage? Vanilla does a flat 1/4th of the damage taken {so you can have an empty slot eat 1/4th of the durability lost} 
But we also want the special armor to be able to say 'Im taking 50% of this damage, no matter what any of the other armor says.'